src/contexts/notification-context.tsx

```diff
@@ -0,0 +1,164 @@
+import React, { createContext, useContext, useState, useEffect, useMemo } from 'react';
+
+import Logger from '../logger/logger';
+import {
+  NotificationSettings,
+  NotificationType,
+  NotificationPreference,
+} from '../types/notification.types';
+import { useLocalStorage } from '../utils';
+
+const DEFAULT_NOTIFICATION_PREFERENCE: NotificationPreference = {
+  enabled: true,
+  sound: true,
+  vibration: true,
+  importance: 'default',
+};
+
+/**
```

Comment:
[How to write better code comments](https://dev.to/adammc331/todo-write-a-better-comment-4c8c)